src/main/io/gps.c

```diff
@@ -474,6 +474,8 @@ void gpsInitHardware(void)
         gpsInitUblox();
 #endif
         break;
+    default:
```

Comment:
Still not quite happy with how this is done here - we are still opening the serial port (if configured) in `gpsInit()` even if the type is `GPS_MSP`. This should be avoided since there is no expectation and no configuration for receiving serial data in this case.

Reply:
How about that in config.c:
```
#if defined(USE_GPS)
    serialPortConfig_t *gpsSerial = findSerialPortConfig(FUNCTION_GPS);
    if(gpsConfig()->provider == GPS_MSP && gpsSerial) {
        serialRemovePort(gpsSerial->identifier);
    }
#endif
    if (
#if defined(USE_GPS)
        gpsConfig()->provider != GPS_MSP && !gpsSerial &&
#endif
        true) {
        featureClear(FEATURE_GPS);
    }
```

Reply:
This is a good idea. But even if this is added, the code that actually opens the serial port should also check for `GPS_MSP` - this will make the code more robust to changes, and make it possible to see that this is how it's meant to work without being aware of the validation and change that is done in `config.c`.

Reply:
OK, here is a snippet how I would change gpsInit():
```
void gpsInit(void)
{
    gpsData.baudrateIndex = 0;
    gpsData.errors = 0;
    gpsData.timeouts = 0;

    memset(gpsPacketLog, 0x00, sizeof(gpsPacketLog));

    // init gpsData structure. if we're not actually enabled, don't bother doing anything else
    gpsSetState(GPS_UNKNOWN);

    gpsData.lastMessage = millis();
    
    if (gpsConfig()->provider == GPS_MSP) { // no serial ports used when GPS_MSP is configured
        return;
    }

    serialPortConfig_t *gpsPortConfig = findSerialPortConfig(FUNCTION_GPS);
    if (!gpsPortConfig) {
        return;
    }
```

Reply:
Yep, this looks good. Thanks.